### ISSUE TITLE
Delay out transitions until new content is ready

### ISSUE THREAD

**Issue by nfourtythree** [gmrchk/swup#46]
Firstly I am loving swup, easy to set up and use it is a great library.

For the project I am currently working on I am trying to get a certain behaviour on transition.

Using JS events I have a loading progress bar at the top which is working nicely. 

What I am trying to achieve is on leaving the page I want to delay the out transitions until the new content is ready. Once it is ready, run and wait for the out transitions of the current content and then run transition in for the new content.

As a process it would be somehthing like:
1. Link Clicked
1. swup starts its background process (no transitions happen)
1. When new content is ready to be replaced - run outbound transitions
1. Run inbound transitions on new content

Hopefully that makes sense. Basically I am trying to avoid the user seeing a "blank page". Let me know if you need any more info or have any questions. Keep up the great work!

**tldr:** delay outbound transitions until new content is ready, then run normal out and in process.
